### PR TITLE
COOPR-604: CLI: help command output should be more readable.

### PR DESCRIPTION
**Pull request by tyatskiv**
@jwang47 I have added the HelpFormatter interface for custom help output, and also implemented the DefaultHelpFormatter for output by default.

Please see updated COOPR help output below:

```
coopr (http://localhost:55054)> help 
COOPR_HOST=localhost
COOPR_USER=admin
COOPR_TENANT=superadmin

Available commands:
add
  • add services <services> on cluster <cluster-id>: Adds one or more services to a cluster
connect
  • connect to <host> as <user> <tenant> [using port <port>] [with ssl <enabled/disabled>]: Connects to a Coopr instance
create
  • create cluster <name> with template <template> of size <size> [using settings <settings>]: Creates a cluster
delete
  • delete cluster <cluster-id> [with provider fields <provider-fields>]: Deletes a cluster
  • delete hardware type <name>: Deletes a hardware type
  • delete image type <name>: Deletes an image type
  • delete provider <name>: Deletes a provider
  • delete resource from automator <automator-id> of type <resource-type> and name <resource-name> and version <resource-version>: Delete a specific version of an automator type resource
  • delete resource from provider <provider-id> of type <resource-type> and name <resource-name> and version <resource-version>: Delete a specific version of a provider type resource
  • delete service <name>: Deletes a service
  • delete template <name>: Deletes a cluster template
  • delete tenant <name>: Deletes a tenant
exit
  • exit: Exits the command-line interface
get
  • get automatortype <automator-id>: Get automator type by id
  • get cluster <cluster-id>: Gets the cluster
  • get cluster-config <cluster-id>: Gets the cluster config
  • get cluster-status <cluster-id>: Gets the cluster status
  • get hardware type <name>: Gets the hardware type
  • get image type <name>: Gets the image type
  • get provider <name>: Gets the provider
  • get providertype <provider-id>: Get provider type
  • get provisioner <provisioner-id>: Gets the provisioner
  • get service <name>: Gets the service
  • get template <name>: Gets the cluster template
  • get tenant <name>: Gets the tenant
help
  • help [<command>]: Prints usage information and description of commands
list
  • list all automatortypes: Lists all automator types
  • list all providertypes: Lists all provider types
  • list clusters: Lists all clusters
  • list hardware types: Lists all cluster hardware types
  • list image types: Lists all image types
  • list providers: Lists all providers
  • list provisioners: Lists all provisioners
  • list resources from automator <automator-id> of type <resource-type> [and status <status>]: Lists automator type resources.
  • list resources from provider <provider-id> of type <resource-type> [and status <status>]: Lists provider type resources.
  • list services: Lists all services
  • list services <cluster-id>: Lists all cluster services
  • list templates: Lists all cluster templates
  • list tenants: Lists all tenants
quit
  • quit: Quits the command-line interface
recall
  • recall resource from automator <automator-id> of type <resource-type> and name <resource-name> and version <resource-version>: Recall a specific version of an automator type resource
  • recall resource from provider <provider-id> of type <resource-type> and name <resource-name> and version <resource-version>: Recall a specific version of a provider type resource.
restart
  • restart service <service-id> on cluster <cluster-id> [with provider fields <provider-fields>]: Restarts service on cluster
set
  • set config <cluster-config> for cluster <cluster-id>: Sets the cluster config
  • set expire time <expire-time> for cluster <cluster-id>: Sets the cluster expire time
stage
  • stage resource from automator <automator-id> of type <resource-type> and name <resource-name> and version <resource-version>: Stage a specific version of an automator type resource
  • stage resource from provider <provider-id> of type <resource-type> and name <resource-name> and version <resource-version>: Stage a specific version of a provider type resource
start
  • start service <service-id> on cluster <cluster-id> [with provider fields <provider-fields>]: Starts service on cluster
stop
  • stop service <service-id> on cluster <cluster-id> [with provider fields <provider-fields>]: Stops service on cluster
sync
  • sync cluster template <cluster-id>: Synchronize the cluster template
  • sync resources: Synchronize resources
```

### CODE REVIEW

**Comment by awholegunch** [comment]
Here is a proposal for changes. I took the copy above and edited it. Best if you diff the two to see changes...

```
coopr (http://localhost:55054)> help 
COOPR_HOST=localhost
COOPR_USER=admin
COOPR_TENANT=superadmin

Available commands:
add
  • add services <services> on cluster <cluster-id>: Add one or more services to a cluster
connect
  • connect to <host> as <user> <tenant> [using port <port>] [with ssl <enabled/disabled>]: Connect to a Coopr instance
create
  • create cluster <name> with template <template> of size <size> [using settings <settings>]: Create a cluster
delete
  • delete cluster <cluster-id> [with provider fields <provider-fields>]: Delete a cluster
  • delete hardware type <name>: Delete a hardware type
  • delete image type <name>: Delete an image type
  • delete provider <name>: Delete a provider
  • delete resource from automator <automator-id> of type <resource-type> and name <resource-name> and version <resource-version>: Delete a specific version of an automator type resource
  • delete resource from provider <provider-id> of type <resource-type> and name <resource-name> and version <resource-version>: Delete a specific version of a provider type resource
  • delete service <name>: Delete a service
  • delete template <name>: Delete a cluster template
  • delete tenant <name>: Delete a tenant
exit
  • exit: Exits the command line interface
get
  • get automator type <automator-id>: Get the automator type by id
  • get cluster <cluster-id>: Get the cluster by id
  • get cluster-config <cluster-id>: Get the cluster config by id
  • get cluster-status <cluster-id>: Get the cluster status by id
  • get hardware type <name>: Get the hardware type
  • get image type <name>: Get the image type
  • get provider <name>: Get the provider
  • get provider type <provider-id>: Get the provider type by id
  • get provisioner <provisioner-id>: Get the provisioner by id
  • get service <name>: Get the service
  • get template <name>: Get the cluster template
  • get tenant <name>: Get the tenant
help
  • help [<command>]: Prints usage information or description of a command
list
  • list automator types: List all automator types
  • list provider types: List all provider types
  • list clusters: List all clusters
  • list hardware types: List all hardware types
  • list image types: List all image types
  • list providers: List all providers
  • list provisioners: List all provisioners
  • list resources from automator <automator-id> of type <resource-type> [and status <status>]: List a specific type of automator type resources
  • list resources from provider <provider-id> of type <resource-type> [and status <status>]: List a specific type of provider type resources
  • list services: List all services
  • list services <cluster-id>: List all cluster services
  • list templates: List all templates
  • list tenants: List all tenants
quit
  • quit: Quits the command line interface
recall
  • recall resource from automator <automator-id> of type <resource-type> and name <resource-name> and version <resource-version>: Recall a specific version of an automator type resource
  • recall resource from provider <provider-id> of type <resource-type> and name <resource-name> and version <resource-version>: Recall a specific version of a provider type resource
restart
  • restart service <service-id> on cluster <cluster-id> [with provider fields <provider-fields>]: Restart a service on a cluster
set
  • set config <cluster-config> for cluster <cluster-id>: Set the cluster config
  • set expire time <expire-time> for cluster <cluster-id>: Set the cluster expire time
stage
  • stage resource from automator <automator-id> of type <resource-type> and name <resource-name> and version <resource-version>: Stage a specific version of an automator type resource
  • stage resource from provider <provider-id> of type <resource-type> and name <resource-name> and version <resource-version>: Stage a specific version of a provider type resource
start
  • start service <service-id> on cluster <cluster-id> [with provider fields <provider-fields>]: Start a service on a cluster
stop
  • stop service <service-id> on cluster <cluster-id> [with provider fields <provider-fields>]: Stop a service on a cluster
sync
  • sync cluster template <cluster-id>: Synchronize the cluster template
  • sync resources: Synchronize resources
```

File: common-cli/src/main/java/co/cask/common/cli/util/DefaultHelpFormatter.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+    String previousCommandName = "";
+    for (Command command : sortedCommands) {
+      String[] commandName = command.getPattern().split(" ", 2);
+      if (previousCommandName.compareTo(commandName[0]) != 0) {
+        previousCommandName = commandName[0];
+        printStream.println(String.format("%s", previousCommandName));
+      }
+
+      printStream.println(String.format("  • %s: %s", command.getPattern(), command.getDescription()));
```
Comment (jwang47) [inline common-cli/src/main/java/co/cask/common/cli/util/DefaultHelpFormatter.java]: Use an asterisk `*` instead of a bullet.

**Comment by jwang47** [comment]
Minor comment otherwise LGTM. @awholegunch we can make those changes in another PR.

**Comment by awholegunch** [comment]
@jwang47 Sure. I'll move this copy to the issue...

**Comment by tyatskiv** [comment]
@jwang47 I have replaced bullet by the asterisk, please take a look.

**Comment by jwang47** [comment]
LGTM

**Pull request merged**